### PR TITLE
update packages

### PR DESCRIPTION
**Pull request by garbas**
Motivation was to update sources of nixpkgs (since we were running quite behind) and to (for now manually) packages for all projects. new project ``releng_treestatus`` was added.

### CODE REVIEW

File: lib/releng_common/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      python.packages."Jinja2"
+    ] ++ optionals (builtins.elem "cache" extras) [ python.packages."Flask-Cache" ]
+      ++ optionals (builtins.elem "db" extras) [ python.packages."Flask-SQLAlchemy" python.packages."Flask-Migrate" ]
+      ++ optionals (builtins.elem "auth" extras) [ python.packages."Flask-Login" python.packages."taskcluster" ]
+      ++ optionals (builtins.elem "api" extras) [ python.packages."connexion" ]
+      ++ optionals (builtins.elem "log" extras) [ python.packages."structlog" ]
+      ++ optionals (builtins.elem "cors" extras) [ python.packages."Flask-Cors" ];
+  checkPhase = ''
+    flake8 --exclude=nix_run_setup.py,migrations/,build/
+    # TODO: py.test
```
Comment (La0) [inline lib/releng_common/default.nix]: So that's where we should run unit tests...

File: lib/releng_common/setup.py [hunk 2]
```diff
@@ -14,18 +14,25 @@
...
+        'Jinja2',
+    ],
+    extras_require=dict(
+        cache=['Flask-Cache'],
+        db=['Flask-SQLAlchemy', 'Flask-Migrate'],
+        auth=['Flask-Login', 'taskcluster'],
+        api=['connexion'],
+        log=['structlog'],
+        cors=['Flask-Cors'],
+    ),
```
Comment (La0) [inline lib/releng_common/setup.py]: The releng_common Flask "plugins" are now optional ?

File: nix/lib/default.nix [hunk 3]
```diff
@@ -279,53 +314,66 @@ in rec {
...
-        propagatedBuildInputs =
-          fromRequirementsFile propagatedRequirements python.packages;
+
+        inherit src;
+
+        buildInputs = [
+          makeWrapper
+          glibcLocales
+          python.packages."flake8"
+          python.packages."gunicorn"
```
Comment (La0) [inline nix/lib/default.nix]: I think there should be Postgresql here, too.

File: nix/lib/default.nix [hunk 4]
```diff
@@ -348,36 +396,32 @@ in rec {
...
           export LOCALE_ARCHIVE=${glibcLocales}/lib/locale/locale-archive
-          for i in src-*; do
-            pushd $i
-            flake8 --exclude=nix_run_setup.py --exclude=build/*
-            # TODO: py.test
-            popd >> /dev/null
-          done
+
+          flake8 --exclude=nix_run_setup.py,migrations/,build/
+          # TODO: py.test
```
Comment (La0) [inline nix/lib/default.nix]: You can add it already, there are dummy unit tests for releng_clobberer and shipit_dashboard

File: nix/nixpkgs.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
-  "owner": "garbas",
-  "repo": "nixpkgs",
-  "rev": "97598ba6707bcd0d6992dcf87ca391326a9c2107",
-  "sha256": "0szd2dgxs30k9f8xfk4lavd2sa7yv6r8ry35iys2sjac50m8crbv"
+  "owner": "NixOS",
```
Comment (La0) [inline nix/nixpkgs.json]: :+1:

File: src/shipit_dashboard/default.nix [hunk 6]
```diff
@@ -5,41 +5,23 @@ let
...
-      ];
-    propagatedRequirements =
-      [ ./../../lib/releng_common/requirements.txt
-        ./requirements.txt
-        ./requirements-prod.txt
+    version = fileContents ./../../VERSION;
+    src = filterSource ./. { inherit name; };
+    propagatedBuildInputs =
+      [ releng_common
+        python.packages."libmozdata"
```
Comment (La0) [inline src/shipit_dashboard/default.nix]: :+1:

File: src/shipit_dashboard/setup.py [hunk 7]
```diff
@@ -11,23 +11,22 @@
...
     name='shipit_dashboard',
-    version=open(os.path.join(here, 'VERSION')).read().strip(),
+    version=version,
     description='The code behind https://dashboard.shipit.mozilla-releng.net',
     author='Mozilla Release Engineering',
     author_email='release@mozilla.com',
     url='https://dashboard.shipit.mozilla-releng.net',
-    install_requires=install_requires,
+    install_requires=[
+        'releng_common[log,api,cors,auth,db]',
```
Comment (La0) [inline src/shipit_dashboard/setup.py]: :clap: Nice, i like what you did with the python extras for releng_common

File: src/shipit_dashboard/setup.py [hunk 8]
```diff
@@ -11,23 +11,22 @@
...
+        # need to pin libmozdata due to
+        # https://github.com/mozilla/libmozdata/commit/e0862e52fa77e38366d64c7068f908362d0535be
+        'libmozdata==0.1.16',
```
Comment (La0) [inline src/shipit_dashboard/setup.py]: Why isn't it in requirements.txt though ?

File: lib/releng_common/default.nix (hunk 1 above)
Comment (garbas) [inline lib/releng_common/default.nix]: yes to enable tests for ``releng_common``

File: lib/releng_common/setup.py (hunk 2 above)
Comment (garbas) [inline lib/releng_common/setup.py]: yes, not every service will use ``db`` or ``auth`` which would mean smaller docker images and faster deployment.

File: nix/lib/default.nix (hunk 4 above)
Comment (garbas) [inline nix/lib/default.nix]: ok i'll try to enable it and also add dummy tests for other services

File: src/shipit_dashboard/setup.py (hunk 8 above)
Comment (garbas) [inline src/shipit_dashboard/setup.py]: good point... would make much more sense there

File: nix/lib/default.nix (hunk 3 above)
Comment (garbas) [inline nix/lib/default.nix]: i think i should add this to ``releng_common``. eg. when extra ``db`` is requested also add ``pysopg2``

**Pull request merged**